tests/chmgt_test.go

```diff
@@ -0,0 +1,222 @@
+package chmgt_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/heimdal-rw/chmgt/config"
+	"github.com/heimdal-rw/chmgt/handling"
+	"github.com/heimdal-rw/chmgt/models"
+)
+
+type App struct {
+	Router http.Handler
+	DB     *models.Datasource
+}
+
+var a App
+
+func TestMain(m *testing.M) {
+	a = App{}
+
+	conf, err := config.ReadConfig("test_config.toml")
+	if err != nil {
+		log.Fatal(err)
+	}
+	handler, err := handling.NewHandler(conf)
+	if err != nil {
+		log.Fatal(err)
+	}
+	handler.Datasource, err = models.NewDatasource(
+		conf.DatabaseConnection(),
+		conf.Database.Name,
+	)
+
+	a.Router = handler.Router
+	a.DB = handler.Datasource
+
+	code := m.Run()
+
+	clearCollections()
+
+	os.Exit(code)
+}
+
+func TestGetSingleUser(t *testing.T) {
+	clearCollections()
+	insertUsers(a.DB, []string{"admin"})
+
+	token, err := getAuthToken("admin", "password_admin")
+	if err != nil {
+		t.Errorf("Got %s when attempting to auth", err)
+	}
+
+	req, _ := http.NewRequest("GET", "/api/users", nil)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
+	response := executeRequest(req)
+
+	checkResponseCode(t, http.StatusOK, response.Code)
```

Comment:
Any specific reason for breaking this out to a separate function and not the other tests?

Reply:
Main reason being we would definitely be re-using this function. I wasn't sure if I'd be re-using the others exactly, but I'd have to write more of the other tests to determine if I should break them out.